dapp/src/multi_runner.rs

```diff
@@ -154,11 +154,11 @@ where
         let results = tests
             .into_iter()
             .map(|(name, contract)| {
-                let address = addresses
+                let (address, init_logs) = addresses
                     .get(name)
                     .ok_or_else(|| eyre::eyre!("could not find contract address"))?;
 
-                let result = self.run_tests(name, contract, *address, &pattern)?;
+                let result = self.run_tests(name, contract, *address, init_logs, &pattern)?;
                 Ok((name.clone(), result))
             })
             .filter_map(|x: Result<_>| x.ok())
```

Comment:
The next line 
```rust
.filter_map(|(name, res)| if res.is_empty() { None } else { Some((name, res)) })
```
is essentially `.filter(|(_, res)| res.is_empty() )` ?

Reply:
yes